app/controllers/tasks_controller.rb

```diff
@@ -16,11 +17,23 @@ def index
       if params[:search][:status].present?
         tasks = tasks.search_status(params[:search][:status])
       end
-      if params[:search][:priority].present?
+      if params[:search][:priority].to_i == 1
         tasks = tasks.order_priority
       end
     end
-    @tasks = tasks.order_created_at.page(params[:page]).per(PER)
+    tasks = tasks.order_created_at
+    if params[:search].present?
+      if params[:search][:label_ids].present?
+        tasks_clone = []
+        tasks.each do |task|
+          if task.label_ids.include?(params[:search][:label_ids].to_i)
+            tasks_clone.push(task)
+          end
```

Comment:
後置ifを使うと1行で記載できますが、見づらいようであれば現状のままでもOKです。
```suggestion  
        tasks_clone.push(task) if task.label_ids.include?(params[:search][:label_ids].to_i)
```